src/cardstack/al-cardstack-view.ts

```diff
@@ -285,41 +323,82 @@ export abstract class AlCardstackView< EntityType=any,
                 caption: properties.caption
             };
         } );
-        this.cards.push( ...newData );
-        this.cards = this.cards.map( c => this.evaluateCardState( c ) );
-        this.visibleCards = this.cards.reduce( ( count, card ) => count + ( card.visible ? 1 : 0 ), 0 );
-
-        if (this.localPagination) {
-            this.markCardsAsCheck();
-        }
-        return this.visibleCards;
+        return newData;
     }
 
     /**
      *  Method to determine visibility of an individual card item based on the current set of active filters.
      */
-    protected evaluateCardState( card:AlCardstackItem<EntityType,PropertyType> ):AlCardstackItem<EntityType,PropertyType> {
-        card.visible = true;
+    protected evaluateCardVisibilityByFilter( card:AlCardstackItem<EntityType,PropertyType> ):boolean {
+        let visible = true;
         let filterProperties = Object.keys( this.activeFilters );
         if ( filterProperties.length === 0 ) {
-            return card;
+            return true;
         }
-        filterProperties.find( property => {
+        filterProperties.forEach( property => {
             if ( ! card.properties.hasOwnProperty( property ) || typeof( ( card.properties as any)[property] ) === 'undefined' ) {
-                card.visible = false;
-                return true;        //  terminate iteration
+                visible = false;
+                return ;        //  terminate iteration
             }
             let cardPropValue = ( card.properties as any )[property];
             let matched = Object.values( this.activeFilters[property] ).find( valDescriptor => {
-                return valDescriptor.value === cardPropValue ? true : false;
+                if (cardPropValue instanceof Array) {
+                    return cardPropValue.includes(valDescriptor.value);
+                }
+                return valDescriptor.value === cardPropValue.value;
             } );
             if ( ! matched ) {
-                card.visible = false;
-                return true;        //  terminate iteration
+                visible = false;
+                return ;        //  terminate iteration
             }
-            return false;
+            return ;
         } );
-        return card;
+        return visible;
+    }
+
+    protected evaluateCardVisibilityBySearch( card:AlCardstackItem<EntityType,PropertyType>, search: RegExp|null):boolean {
+        let visible = false;
+        if (search === null) {
+            return true;
+        }
+
+        if(this.characteristics && this.characteristics.searchableBy) {
+            if (this.characteristics.searchableBy.length === 0 ) {
+                return true;
+            }
+            this.characteristics.searchableBy.forEach( (property:string) => {
```

Comment:
Again with `filter` instead of `forEach` -- as soon as we have a match, we can stop iterating.